simulator/virtual_machine.go

```diff
@@ -1888,10 +1888,15 @@ func (vm *VirtualMachine) customize(ctx *Context) {
 		hostname = customizeName(vm, c.UserData.ComputerName)
 	}
 
+	cards := object.VirtualDeviceList(vm.Config.Hardware.Device).SelectByType((*types.VirtualEthernetCard)(nil))
+
 	for i, s := range vm.imc.NicSettingMap {
 		nic := &vm.Guest.Net[i]
 		if s.MacAddress != "" {
-			nic.MacAddress = s.MacAddress
+			nic.MacAddress = strings.ToLower(s.MacAddress) // MacAddress in guest will always be lowercase
+			card := cards[i].(types.BaseVirtualEthernetCard).GetVirtualEthernetCard()
```

Comment:
`cards` slice would always match the number of NIC mappings, i.e. nil pointer exception or cards to settings mismatch would never occur correct?

Reply:
Right, we validate the length as part of `CustomizeVM` here: https://github.com/vmware/govmomi/blob/570383b878d4accd83863ec42e436ad7bec5a377/simulator/virtual_machine.go#L1956-L1961